### PR TITLE
refactor(ckbtc): use IC CDK to interact with Bitcoin canister

### PR DESCRIPTION
**Pull request by lpahlavi**
([XC-229](https://dfinity.atlassian.net/browse/XC-229)) Interact with the Bitcoin canister through the IC CDK instead of the management canister within the ckBTC minter.

[XC-229]: https://dfinity.atlassian.net/browse/XC-229?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lpahlavi** [comment]
Thank you for the review @gregorydemay, I've incorporated your feedback!

File: rs/bitcoin/mock/src/main.rs [hunk 1]
```diff
@@ -238,3 +242,17 @@ fn check_candid_interface_compatibility() {
         candid_parser::utils::CandidSource::File(old_interface.as_path()),
     );
 }
+
+fn assert_network_equivalent(lhs: &NetworkInRequest, rhs: &Network) {
+    match lhs {
```
Comment (ninegua) [inline rs/bitcoin/mock/src/main.rs:247]: nit: suggest `assert_eq!(Network::from(*lhs), *rhs)`

**Review by ninegua — APPROVED** [review]
Nice! Made some minor suggestions

**Comment by lpahlavi** [comment]
@ninegua Thank a lot for your feedback! I'd missed the built-in conversion between Network and NetworkInRequest, looks much better now :)

**Pull request merged**